### PR TITLE
[5.2] Implemented job:dispatch and job:list artisan commands.

### PR DESCRIPTION
**Pull request by 20TRIES**
I've created two artisan commands:
- job:list; lists all available jobs.
- job:dispatch; dispatches a job.

I've found them to be quite useful in the past to test / run jobs from the command line.

I've also used it with the scheduler to schedule jobs as shown below:

`$schedule->command('job:dispatch SomeJobClass')`

### CODE REVIEW

**Comment by taylorotwell** [comment]
I think job:dispatch is really cool but not sure about job:list. Jobs could be spread around all across the project and not necessarily in that one directory.

**Comment by 20TRIES** [comment]
Yer i see what you mean... the main reason i went for the list command was because it makes it a bit easier finding the job name; otherwise i found that myself having to find the file to get the namespace and class name.

I could remove it, or just make it possible to dispatch from a full namespace incase the job class exists outside of the default location?

**Comment by taylorotwell** [comment]
I would remove job:list but job:dispatch is cool for sure.

**Comment by 20TRIES** [comment]
All sorted.

File: src/Illuminate/Foundation/Providers/ConsoleSupportServiceProvider.php [hunk 1]
```diff
@@ -25,5 +26,6 @@ class ConsoleSupportServiceProvider extends AggregateServiceProvider
         'Illuminate\Database\SeedServiceProvider',
         'Illuminate\Foundation\Providers\ComposerServiceProvider',
         'Illuminate\Queue\ConsoleServiceProvider',
+        JobsServiceProvider::class,
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Providers/ConsoleSupportServiceProvider.php]: Please use the same notation here as the others.

File: src/Illuminate/Console/Jobs/DispatchCommand.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    protected $signature = 'job:dispatch {job : The name of the job that should be dispatched}';
+
+    /**
+     * Execute the console command.
+     */
+    public function fire()
+    {
+        $job = $this->argument('job');
+
+        $this->info("Dispatching $job now...");
```
Comment (GrahamCampbell) [inline src/Illuminate/Console/Jobs/DispatchCommand.php]: Please use braces around the variable. Although it's not necessary, Laravel normally does it for clarity.

**Comment by GrahamCampbell** [comment]
Surely most jobs take constructor parameters? We can't support that here, so this command has little use IMO. It's just a simple to go into artisan tinker.

**Comment by 20TRIES** [comment]
Some of the jobs we use take parameters, others, especially those that are scheduled, do not.

So yer, the main use case i guess would be testing these, scheduling them and manually running them.

Thats cool if you don't think its necessary though; just thought i'd see what you guys thought :)

File: src/Illuminate/Foundation/Providers/ConsoleSupportServiceProvider.php (hunk 1 above)
Comment (20TRIES) [inline src/Illuminate/Foundation/Providers/ConsoleSupportServiceProvider.php]: Is it worth swapping them all to that notation? It works really nice when refactoring apposed to using string class names?
I've seen it in other parts of Laravel that way, is there a preference as to which notation should be used?

File: src/Illuminate/Console/Jobs/DispatchCommand.php (hunk 2 above)
Comment (20TRIES) [inline src/Illuminate/Console/Jobs/DispatchCommand.php]: No probs, will sort this in the morning.

File: src/Illuminate/Console/Jobs/DispatchCommand.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+class DispatchCommand extends Command
+{
+    use DispatchesJobs;
+
+    /**
+     * The console command name.
+     *
+     * @var string
+     */
+    protected $name = 'job:dispatch';
```
Comment (lucasmichot) [inline src/Illuminate/Console/Jobs/DispatchCommand.php]: This should be removed : the command `$signature` is already there

File: src/Illuminate/Console/Jobs/DispatchCommand.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Dispatches a job.';
+
+    /**
+     * @var string The name and signature of the console command.
```
Comment (lucasmichot) [inline src/Illuminate/Console/Jobs/DispatchCommand.php]: Maybe add `The console command signature.` there ?

File: src/Illuminate/Console/Jobs/DispatchCommand.php [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+     */
+    protected $signature = 'job:dispatch {job : The name of the job that should be dispatched}';
+
+    /**
+     * Execute the console command.
```
Comment (lucasmichot) [inline src/Illuminate/Console/Jobs/DispatchCommand.php]: Missing `@return void`

File: src/Illuminate/Console/Jobs/DispatchCommand.php [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+class DispatchCommand extends Command
+{
+    use DispatchesJobs;
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Dispatches a job.';
```
Comment (jmarcher) [inline src/Illuminate/Console/Jobs/DispatchCommand.php]: Command descriptions do not end with a point.

**Comment by taylorotwell** [comment]
Going to hold off on this.

**Pull request closed without merging**